Use larger of PATH_MAX and page size when allocating buffer.

===================================================================
RCS file: /cvs/glibc/libc/sysdeps/unix/sysv/linux/getcwd.c,v
retrieving revision 1.23
retrieving revision 1.24
diff -u -r1.23 -r1.24
--- libc/sysdeps/unix/sysv/linux/getcwd.c       2006/01/08 08:21:15     1.23
+++ libc/sysdeps/unix/sysv/linux/getcwd.c       2006/04/02 17:58:28     1.24
@@ -87,7 +87,7 @@
          return NULL;
        }
 
-      alloc_size = PATH_MAX;
+      alloc_size = MAX (PATH_MAX, __getpagesize ());
     }
 
   if (buf == NULL)
